src/endpoint/s3/s3_rest.js

```diff
@@ -229,16 +229,22 @@ async function authorize_request_policy(req) {
     }
 
     const account = req.object_sdk.requesting_account;
-    const account_identifier = req.object_sdk.nsfs_config_root ? account.name.unwrap() : account.email.unwrap();
-    const is_system_owner = account_identifier === system_owner.unwrap();
+    const account_identifier_name = req.object_sdk.nsfs_config_root ? account.name.unwrap() : account.email.unwrap();
+    const account_identifier_id = req.object_sdk.nsfs_config_root ? account._id : undefined;
+    //In old buckets (until 5.17), system_owner is account name (sensitive string).
+    //In new buckets (since 5.18), system_owner is account id (string).
+    const system_owner_equatable = system_owner.unwrap ? system_owner.unwrap() : system_owner;
```

Comment:
Are you sure about this line? `system_owner.unwrap`? (did you mean the function `unwrap()`?)
Are you trying to check if it is an ID or a sensitive string?

Reply:
Yes. Sometimes it is a sensitive string, but in nc it's an id.
I think system owner is going to be removed anyway, no?